### ISSUE TITLE
Cannot replay from history

### ISSUE THREAD

**Issue by Donkey-Doug** [KieronQuinn/AmbientMusicMod#135]
Device: Xiaomi Mi 9T Pro running LineageOS based on android 12 without google play services.

I was able to listen once when I first installed ambient music mod, but after that it crashes every time. See video below:

https://user-images.githubusercontent.com/50382414/210181412-12321214-de9a-4807-a499-5e4079f8dca9.mp4

**Comment by KieronQuinn** [comment]
Here's an article on making logs: https://www.xda-developers.com/how-to-take-logs-android/

A system log is needed in this case

**Comment by Donkey-Doug** [comment]
Hi, let me know if this is the good log:

[log for Ambient Music Mod.txt](https://github.com/KieronQuinn/AmbientMusicMod/files/10429415/log.for.Ambient.Music.Mod.txt)

**Comment by KieronQuinn** [comment]
That's absolutely perfect, thank you

```
01-03 12:19:45.276  5507  5507 E AndroidRuntime: FATAL EXCEPTION: main
01-03 12:19:45.276  5507  5507 E AndroidRuntime: Process: com.kieronquinn.app.pixelambientmusic, PID: 5507
01-03 12:19:45.276  5507  5507 E AndroidRuntime: java.lang.IllegalArgumentException: Unknown component: ComponentInfo{com.google.android.googlequicksearchbox/com.google.android.apps.gsa.staticplugins.opa.EnterOpaActivityFromDeeplink}
01-03 12:19:45.276  5507  5507 E AndroidRuntime: 	at android.os.Parcel.createExceptionOrNull(Parcel.java:2430)
01-03 12:19:45.276  5507  5507 E AndroidRuntime: 	at android.os.Parcel.createException(Parcel.java:2410)
01-03 12:19:45.276  5507  5507 E AndroidRuntime: 	at android.os.Parcel.readException(Parcel.java:2393)
01-03 12:19:45.276  5507  5507 E AndroidRuntime: 	at android.os.Parcel.readException(Parcel.java:2335)
01-03 12:19:45.276  5507  5507 E AndroidRuntime: 	at android.content.pm.IPackageManager$Stub$Proxy.getComponentEnabledSetting(IPackageManager.java:7204)
01-03 12:19:45.276  5507  5507 E AndroidRuntime: 	at android.app.ApplicationPackageManager.getComponentEnabledSetting(ApplicationPackageManager.java:2814)
01-03 12:19:45.276  5507  5507 E AndroidRuntime: 	at nij.c(PG:3)
01-03 12:19:45.276  5507  5507 E AndroidRuntime: 	at nnf.a(PG:54)
01-03 12:19:45.276  5507  5507 E AndroidRuntime: 	at abd.l(PG:4)
01-03 12:19:45.276  5507  5507 E AndroidRuntime: 	at abd.c(PG:4)
01-03 12:19:45.276  5507  5507 E AndroidRuntime: 	at abd.j(PG:2)
01-03 12:19:45.276  5507  5507 E AndroidRuntime: 	at abe.j(Unknown Source:0)
01-03 12:19:45.276  5507  5507 E AndroidRuntime: 	at aba.run(PG:90)
01-03 12:19:45.276  5507  5507 E AndroidRuntime: 	at android.os.Handler.handleCallback(Handler.java:938)
01-03 12:19:45.276  5507  5507 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:99)
01-03 12:19:45.276  5507  5507 E AndroidRuntime: 	at android.os.Looper.loopOnce(Looper.java:201)
01-03 12:19:45.276  5507  5507 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:288)
01-03 12:19:45.276  5507  5507 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:7870)
01-03 12:19:45.276  5507  5507 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
01-03 12:19:45.276  5507  5507 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
01-03 12:19:45.276  5507  5507 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1003)
01-03 12:19:45.276  5507  5507 E AndroidRuntime: Caused by: android.os.RemoteException: Remote stack trace:
01-03 12:19:45.276  5507  5507 E AndroidRuntime: 	at com.android.server.pm.PackageManagerService.getComponentEnabledSettingInternal(PackageManagerService.java:24595)
01-03 12:19:45.276  5507  5507 E AndroidRuntime: 	at com.android.server.pm.PackageManagerService.getComponentEnabledSetting(PackageManagerService.java:24578)
01-03 12:19:45.276  5507  5507 E AndroidRuntime: 	at android.content.pm.IPackageManager$Stub.onTransact(IPackageManager.java:3307)
01-03 12:19:45.276  5507  5507 E AndroidRuntime: 	at com.android.server.pm.PackageManagerService.onTransact(PackageManagerService.java:8903)
01-03 12:19:45.276  5507  5507 E AndroidRuntime: 	at android.os.Binder.execTransactInternal(Binder.java:1184)
```

You don't have the Google app installed, that's why. The Now Playing History screen assumes you do, because Google never expects you to uninstall it. I can probably patch something to default to not showing the button.

**Comment by Donkey-Doug** [comment]
The strange thing is that it worked for the first few days after installing.

Since the files are all stored locally, I assume playback using any media player should be doable.

**Comment by KieronQuinn** [comment]
It usually offers other apps like YouTube, YouTube Music, Spotify, Deezer etc. Local playback isn't possible since it doesn't make the link between a recognised track and those on the device, although that might be possible with some ID3 trickery (maybe a good idea for the custom history UI if I ever built that)

**Comment by Donkey-Doug** [comment]
By the way, this log is only for com.kieronquinn.ambientmusicmod.com.

Do you also need a full log? Or maybe also a log for com.kieronquinn.pixelambientmusic?

**Comment by KieronQuinn** [comment]
This should be fixed in 2.1, let me know if it still happens.
